rust/src/pgsql/pgsql.rs

```diff
@@ -164,6 +173,17 @@ impl PgsqlState {
             backend_pid: 0,
             state_progress: PgsqlStateProgress::IdleState,
             tx_index_completed: 0,
+            events: 0,
+        }
+    }
+
+    /// Set an event on the most recent transaction
+    fn set_event(&mut self, event: PgsqlEvent) {
```

Comment:
Can you write this in `impl PgsqlTransaction ` ?

As it belongs to a transaction, not the state...

Reply:
Connected to my wondering of what to do in case the event is the result of a parsing error, and to the possibility of not existing a transaction, I see that for there's a `set_event_notx` function (https://github.com/OISF/suricata/commit/1ba62993d5fc6ac6fa8). Wondering if this could be the answer.

Reply:
I may create a transaction for the sole purpose of having the event...
Not sure it is the best way...